### ISSUE TITLE
Message Operation: Batch Permute Parameter Shape Mismatch

### ISSUE THREAD

**Issue by wangyz1999** [TownesZhou/PySigma#12]
```python
def test_parameter_batch_permute(self):
    m1 = Message(MessageType.Parameter,
                 batch_shape=torch.Size([2, 3, 4]),
                 param_shape=torch.Size([2]),
                 parameters=torch.ones(2, 3, 4, 2)
                 )
    m2 = m1.batch_permute([2, 0, 1])
```

In this case, `pos_target_dims == [2, 0, 1]`
`b_p_dims = pos_target_dims + [0] == [2, 0, 1, 0]`

`new_parameters = self.parameters.permute(b_p_dims)` gives runtime error:
repeated dim in permute

https://github.com/TownesZhou/PySigma/blob/47c9c5ab3143b2a1092721a982644d31ff4f50b1/pysigma/defs.py#L409-L449
